### PR TITLE
Undo + Restart options

### PR DESCRIPTION
**Pull request by Gurveer90**
I think it is good to add Restart in case player do not want Undo multiple times

### CODE REVIEW

File: sokoban.hs [hunk 1]
```diff
@@ -198,55 +202,68 @@ play (QuitGame s) =
   do
     putStrLn "Exiting level..."
 
+play (UndoGame (State board pS (r, c) l)) =
+  do
+    putStrLn "Switching to previous move..."
+    if pS == Empty then (putStrLn "No previos move...")
```
Comment (jsondoo) [inline sokoban.hs]: Typo "previous"

File: sokoban.hs [hunk 2]
```diff
@@ -198,55 +202,68 @@ play (QuitGame s) =
...
+getNextBoard :: Game 
+getNextBoard move (State board pS (r,c) l)
+  | move == (Action 'W') = movePlayer (State board (State board pS (r,c) l) (r,c) l) (r-1,c) (r-2,c)
+  | move == (Action 'A') = movePlayer (State board (State board pS (r,c) l) (r,c) l) (r,c-1) (r,c-2)
+  | move == (Action 'S') = movePlayer (State board (State board pS (r,c) l) (r,c) l) (r+1,c) (r+2,c)
+  | move == (Action 'D') = movePlayer (State board (State board pS (r,c) l) (r,c) l) (r,c+1) (r,c+2)
+  | move == (Action 'Q') = QuitGame (State board (State board pS (r,c) l) (r, c) l)
+  | move == (Action 'R') = RestartGame (State board (State board pS (r,c) l) (r, c) l)
+  | move == (Action 'U') = UndoGame (State board pS (r, c) l)
+  | otherwise = ContinueGame (State board (State board pS (r,c) l) (r, c) l) -- return same state
```
Comment (jsondoo) [inline sokoban.hs]: Might be cleaner if we define 
`where newPrevState = (State board pS (r,c) l)`

**Comment by jsondoo** [comment]
Good work :+1:

File: sokoban.hs (hunk 1 above)
Comment (Gurveer90) [reply sokoban.hs]: ok

File: sokoban.hs (hunk 2 above)
Comment (Gurveer90) [reply sokoban.hs]: ok

**Comment by Gurveer90** [comment]
Will modify after merge

**Pull request merged**